sklearn_numba_dpex/common/matmul.py

```diff
@@ -195,18 +195,8 @@ def out_fused_elementwise_fn(x):
 
     # Such a work group loads two sliding windows that span relevant rows and colums
     # of the inputs X and Y_t.
-    local_X_sliding_window_shape = (
-        result_window_height,
-        2 * sub_group_size,  # allocate twice the space to let padding against bank
-        # conflicts
-    )
-    local_Y_t_sliding_window_shape = (
-        result_window_width,
-        2 * sub_group_size,  # allocate twice the space to let padding against bank
-        # conflicts
-    )
-    # TODO: it's surprising to see bank conflicts here, however the padding add
-    # 10% better performance on iGPUs. Why ?
+    local_X_sliding_window_shape = (sub_group_size, result_window_height)
+    local_Y_t_sliding_window_shape = (sub_group_size, result_window_width)
```

Comment:
NB: this PR also transposes the layout of the sliding windows. I don't understand why, but it seems to be half the reason we've unlocked better performance. :shrug: